libyul/ObjectParser.cpp

```diff
@@ -104,9 +111,62 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::SourceNameMap> ObjectParser::tryParseSourceNameMapping() const
+{
+	// @use-src 0:"abc.sol", 1:"foo.sol", 2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		"(^|\\s)@use-src\\s((.|[\r\n])*)",
```

Comment:
As I said, this is not enough. Better safe than sorry and use something like
```suggestion
		"(^|\\s)@use-src\\s((x|[^x])*)",
```

Reply:
Also the `\\s` at the end has to be optional:
```suggestion
		"(^|\\s)@use-src(\\s(x|[^x])*)?",
```

Reply:
@chriseth um, I just pushed an alternative approach by not even trying to match the `@use-src`'s right-hand-side.